### ISSUE TITLE
backend: export methods

### ISSUE THREAD

**Issue by gucio321** [AllenDang/cimgui-go#169]
Here are backend methods that need to be exported to go
- [X] #192
- [X] SetCloseCallback
- [X] SetIcon
- [X] SetSizeLimit
- [X] SetTitle
- [X] SetShouldClose
- [ ] SetKeyCallback
- [ ] SetSizeChangeCallback

**Comment by Chillance** [comment]
I wanted to help out so I took a quick look but seems a bit more involved. However, I noticed some things might even be in place such as:
```
func (b GLFWBackend) SetShouldClose(value bool) {
	C.igGLFWWindow_SetShouldClose(b.handle(), C.int(castBool(value)))
}

```
However, not sure how "platform" would be in gui...

**Comment by AllenDang** [comment]
It's easy, just expose them from GLFW, I could do it after this weekend.

**Comment by gucio321** [comment]
> It's easy, just expose them from GLFW, I could do it after this weekend.

It'd be great. I'm on holiday the next week so i will be able to finish this as i comr back home

**Comment by Wolf65** [comment]
Is it possible to export glfw.WindowHint methods? I would like to be able to turn off the main window so that subwindows are on top of all windows.

**Comment by AllenDang** [comment]
> Is it possible to export glfw.WindowHint methods? I would like to be able to turn off the main window so that subwindows are on top of all windows.

Yeah, it's possible, I will spend some time later to implement that.

**Comment by gucio321** [comment]
@AllenDang could you merge #170 pls

**Comment by gucio321** [comment]
@AllenDang one more method we need is `GetPos`

**Comment by gucio321** [comment]
- @Wolf65 `WindowHint` will be here soon: https://github.com/AllenDang/cimgui-go/pull/190

**Comment by Wolf65** [comment]
@gucio321 Cool! But then there is a question now when creating ```backend.CreateWindow("Hello from cimgui-go", 1200, 900, 0)```.
The last argument is responsible for the flags that in the file glfw_backend.cpp on lines 73-91 in fact and call hint. And now before creating a window we define hint and in creating a window the hint part will be repeated.

**Comment by gucio321** [comment]
I see, but sadly can't do anything with it atm.
Maybe will take a look later

**Comment by gucio321** [comment]
> @gucio321 Cool! But then there is a question now when creating backend.CreateWindow("Hello from cimgui-go", 1200, 900, 0).
The last argument is responsible for the flags that in the file glfw_backend.cpp on lines 73-91 in fact and call hint. And now before creating a window we define hint and in creating a window the hint part will be repeated.

@Wolf65 could you please open a separted issue for this?
This one will be fixed in #193

**Comment by Wolf65** [comment]
@gucio321 open
